test/framework/src/main/java/org/elasticsearch/script/MockScriptService.java

```diff
@@ -39,4 +43,30 @@ public MockScriptService(Settings settings, Map<String, ScriptEngine> engines, M
     boolean compilationLimitsEnabled() {
         return false;
     }
+
+    public static <T> MockScriptService singleContext(ScriptContext<T> context, Function<String, T> compile,
```

Comment:
should the other test code use this too?

Reply:
It can, but I wanted to keep this PR small. We can migrate other uses to this later as necessary.

Reply:
👍 was mostly wondering, as it seemed convenient